src/common/FPSTimer.cpp

```diff
@@ -57,3 +57,25 @@ void FPSTimer::update(double elapsedTime, double renderingTime, int logCount)
     mRecordFpsFrequencyCursor++;
 }
 
+bool FPSTimer::variance(float *avg)
+{
+    for (int i = 0; i < NUM_HISTORY_DATA; i++)
+    {
+        *avg += mHistoryFPS[i];
+    }
+    *avg /= NUM_HISTORY_DATA;
+
+    float var = 0.f;
+    for (int i = 0; i < NUM_HISTORY_DATA; i++)
+    {
+        var += pow(mHistoryFPS[i] - *avg, 2);

```

Comment:
How about removing the history data with big varience and recalculate the average value again instead of discarding the whole data?

Reply:
The history fps records latest 100 data. Old data is discarded during rendering.